packages/block-editor/src/components/block-list/use-block-props/use-block-refs.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	useCallback,
+	useContext,
+	useLayoutEffect,
+	useMemo,
+	useRef,
+	useState,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { BlockRefs } from '../../provider/block-refs-provider';
+
+/** @typedef {import('@wordpress/element').RefCallback} RefCallback */
+/** @typedef {import('@wordpress/element').RefObject} RefObject */
+
+/**
+ * Provides a ref to the BlockRefs context.
+ *
+ * @param {string} clientId The client ID of the element ref.
+ *
+ * @return {RefCallback} Ref callback.
+ */
+export function useBlockRefProvider( clientId ) {
+	const { refs, callbacks } = useContext( BlockRefs );
+	const ref = useRef();
+	useLayoutEffect( () => {
+		refs.set( clientId, ref );
+		return () => {
+			refs.delete( clientId );
+		};
+	}, [] );
+	return useCallback( ( element ) => {
+		// Update the ref in the provider.
+		ref.current = element;
```

Comment:
I wonder if we could just set the element in the context here instead of using an intermediary "ref". I thought the "ref" was useful for useBlockRef, but it doesn't seem to be the case anyway.

Reply:
We could through `useMergeRefs`, but we'd need to callback anyway as well, so this seems a bit simpler.